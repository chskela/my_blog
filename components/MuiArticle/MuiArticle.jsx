import React from "react";
import cn from "classnames";
import Link from "next/link";
import PropTypes from "prop-types";

import classes from "./class.module.css";

import { formatDate } from "@lib/formatDate";
import MuiCardDescription from "@components/MuiCardDescription/MuiCardDescription";

const MuiArticle = React.memo(({ post, classNames }) => {
  const { title, author, date, excerpt, category, slug } = post;
  const postDate = formatDate(date);

  return (
    <Link href={`/posts/${slug}`}>
      <section className={cn(classes.wrapper, classNames?.wrapper)}>
        <div className={cn(classes.content, classNames?.content)}>
          <MuiCardDescription
            category={category}
            title={title}
            author={author}
            date={postDate}
            excerpt={excerpt}
          />
        </div>
      </section>
    </Link>
  );
});

MuiArticle.propTypes = {
  post: PropTypes.shape({
    title: PropTypes.string.isRequired,
    author: PropTypes.shape({
      name: PropTypes.string.isRequired,
      picture: PropTypes.string.isRequired,
    }),
    date: PropTypes.string.isRequired,
    excerpt: PropTypes.string.isRequired,
    category: PropTypes.string.isRequired,
  }),
  className: PropTypes.shape({
    wrapper: PropTypes.string,
    content: PropTypes.string,
  }),
};

MuiArticle.defaultProps = {
  className: {
    wrapper: "",
    content: "",
  },
};

export default MuiArticle;
