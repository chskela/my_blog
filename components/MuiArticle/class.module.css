.wrapper {
  background: linear-gradient(
      0deg,
      rgba(28, 28, 28, 0.1),
      rgba(28, 28, 28, 0.1)
    ),
    url(/assets/images/hero.jpeg) center no-repeat;
  background-size: cover;
  min-height: 592px;
  width: 100%;
}

.content {
  background-color: var(--color-primary);
  max-width: 509px;
  margin-left: var(--size-12);
  padding: var(--size-12) var(--size-10) var(--size-16);
}
