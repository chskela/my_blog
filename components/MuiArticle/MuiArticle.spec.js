import React from "react";
import { shallow } from "enzyme";
import MuiArticle from "./MuiArticle";

const setUp = (props) => shallow(<MuiArticle {...props} />);

describe("MuiArticle component", () => {
  const props = {
    post: {
      title: "test_title",
      author: {
        name: "test_name",
        picture: "test_picture",
      },
      date: "test_date",
      excerpt: "test_excerpt",
      category: "test_category",
    },
  };

  it("should render MuiArticle component default props", () => {
    const component = setUp(props);
    expect(component).toMatchSnapshot();
  });
  it("should render MuiArticle component whis props", () => {
    const component = setUp({
      ...props,
      className: {
        wrapper: "test_wrapper",
        content: "test_content",
      },
    });
    expect(component).toMatchSnapshot();
  });
});
