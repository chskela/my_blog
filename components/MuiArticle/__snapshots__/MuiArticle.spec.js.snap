// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MuiArticle component should render MuiArticle component default props 1`] = `
<Link
  href="/posts/undefined"
>
  <section
    className="wrapper"
  >
    <div
      className="content"
    >
      <Memo()
        author={
          Object {
            "name": "test_name",
            "picture": "test_picture",
          }
        }
        category="test_category"
        date="test_date"
        excerpt="test_excerpt"
        title="test_title"
      />
    </div>
  </section>
</Link>
`;

exports[`MuiArticle component should render MuiArticle component whis props 1`] = `
<Link
  href="/posts/undefined"
>
  <section
    className="wrapper"
  >
    <div
      className="content"
    >
      <Memo()
        author={
          Object {
            "name": "test_name",
            "picture": "test_picture",
          }
        }
        category="test_category"
        date="test_date"
        excerpt="test_excerpt"
        title="test_title"
      />
    </div>
  </section>
</Link>
`;
